### PR TITLE
fix: migration queries

### PR DESCRIPTION
**Pull request by bradleystachurski**

### CODE REVIEW

File: src/federation/observer.rs [hunk 1]
```diff
@@ -77,23 +77,36 @@ impl FederationObserver {
...
             &self.connection().await?,
-            // language=postgresql
-            "SELECT
-                    CASE
-                        WHEN EXISTS (
-                            SELECT FROM information_schema.tables
-                            WHERE table_schema = 'public'
-                            AND table_name = 'schema_version'
-                        )
-                        THEN (SELECT COALESCE(MAX(version), 0) FROM schema_version)
```
Comment (bradleystachurski) [inline src/federation/observer.rs:90]: I'm getting the following error when running the migrations. It looks like postgres still runs subqueries in the `THEN` branch even if `WHEN EXISTS` is false. Creating a function to handle works on my end.

```
Error: db error: ERROR: relation "schema_version" does not exist

Caused by:
    ERROR: relation "schema_version" does not exist
```

File: src/federation/observer.rs [hunk 2]
```diff
@@ -107,12 +120,10 @@ impl FederationObserver {
...
-                self.connection()
-                    .await?
-                    .transaction()
-                    .await?
-                    .batch_execute(migration)
-                    .await?;
+                let mut conn = self.connection().await?;
+                let transaction = conn.transaction().await?;
+                transaction.batch_execute(migration).await?;
+                transaction.commit().await?;
```
Comment (bradleystachurski) [inline src/federation/observer.rs:126]: I might be misreading the api, but I think we need to commit the transaction once we're finished

Comment (elsirion) [reply src/federation/observer.rs:126]: You are very much correct :see_no_evil: I was doing a lot of manual experimentation with the DB schema, so didn't notice a lot of the problems with automated migrations till I deployed to production and then didn't get around updating the PR after fixing things manually.

File: src/federation/observer.rs (hunk 1 above)
Comment (elsirion) [reply src/federation/observer.rs:90]: Tested your version and it works

**Review by elsirion — APPROVED** [review]

**Pull request merged**